### PR TITLE
Refactor lint rendering to avoid loops

### PR DESCRIPTION
**Pull request by kuhnroyal**
This makes future changes to the output easier and allows for easier chaining of more sort parameters (severity).

There are no changes in tests, the output stays the same with this change.

### CODE REVIEW

**Comment by kuhnroyal** [comment]
Not sure what failed there in the format check.

File: packages/custom_lint/lib/custom_lint.dart [hunk 1]
```diff
@@ -82,46 +83,42 @@ Future<void> _runPlugins(
     exitCode = 1;
     stderr.writeln('$err\n$stack');
   }
-
-  // Since no problem happened, we print a message saying everything went well
-  if (exitCode == 0) {
```
Comment (rrousselGit) [inline packages/custom_lint/lib/custom_lint.dart:87]: Why is this deleted?

Comment (rrousselGit) [reply packages/custom_lint/lib/custom_lint.dart:87]: Sorry I missed it in the changes further down

File: packages/custom_lint/lib/custom_lint.dart [hunk 2]
```diff
@@ -82,46 +83,42 @@ Future<void> _runPlugins(
...
-      if (columnCompare != 0) return columnCompare;
-
-      final codeCompare = a.code.compareTo(b.code);
-      if (codeCompare != 0) return codeCompare;
-
-      return a.message.compareTo(b.message);
-    });
+  final errors = lints
+      .expand(
+        (lintsForFile) => lintsForFile.errors
```
Comment (rrousselGit) [inline packages/custom_lint/lib/custom_lint.dart:94]: I don't like the syntax. Too much going on at once.

Why sort the errors twice? Why not have the `..sort` logic in `sortedBy`?

Comment (kuhnroyal) [reply packages/custom_lint/lib/custom_lint.dart]: Good point, I refactored it and now also don't need the `collection` package.

File: packages/custom_lint/lib/custom_lint.dart [hunk 3]
```diff
@@ -89,24 +88,24 @@ void _renderLints(
...
-            if (columnCompare != 0) return columnCompare;
-
-            final codeCompare = a.code.compareTo(b.code);
-            if (codeCompare != 0) return codeCompare;
-
-            return a.message.compareTo(b.message);
-          }),
-      )
-      .sortedBy((e) => _relativeFilePath(e.location.file, workingDirectory));
+  final errors = lints.expand((lintsForFile) => lintsForFile.errors).toList()
```
Comment (rrousselGit) [inline packages/custom_lint/lib/custom_lint.dart:91]: Let's keep using .sorted if we're going to use `toList()`.

But we can do it in two lines if that's more readable.

File: packages/custom_lint/lib/custom_lint.dart (hunk 2 above)
Comment (kuhnroyal) [reply packages/custom_lint/lib/custom_lint.dart]: I can extract the `expand(...)` into a new variable if you prefer this.

File: packages/custom_lint/lib/custom_lint.dart (hunk 3 above)
Comment (kuhnroyal) [reply packages/custom_lint/lib/custom_lint.dart]: Done

**Comment by rrousselGit** [comment]
LGTM~

**Comment by kuhnroyal** [comment]
I think the line number in the golden is now wrong, is was only wrong for main SDK but not for stable which I changed it to.

**Comment by rrousselGit** [comment]
I reverted to master. Didn't see that you switched to stable, which I don't want to do.

**Comment by kuhnroyal** [comment]
Should it not at least run on stable as well to shows when it breaks there due to newer features being used or something?

**Comment by rrousselGit** [comment]
No. It's not an issue anyway
